### ISSUE TITLE
[Question] How can I write api response using interceptor into file and attach it in test.step() 

### ISSUE THREAD

**Issue by Mictim** [microsoft/playwright#10935]
Hi,

I faced a problem with the implementation of the common steps when I try to intercept the response from API request and write the result (JSON) to the file.
There is an example of the code:
```
import { test } from "@playwright/test"

export class BaseSteps {

  async writeResponseToJson(id: string) {
    await test.step('Get response and write it to JSON', async() => {
      const [response] = await Promise.all([
        this.page.waitForResponse('**/api/{id}'),
        this.somePage.clickApproveBtn();
      ]);
      const responseBoby = await response.json();
      const buffer: Buffer = Buffer.from(JSON.stringify(responseBody), 'utf8');
      // Here I used in the test body this kind of implementation:
     // But in this implementation should I pass testInfo:TestInfo as well?
      testInfo.attachments.push({name: "Response", path: "response.json", 
         contentType: "application/json",
         body: responseBody});
    });
  }
}
```

Could you please advise how can I resolve this without passing testInfo in each method?
I found this discussion: [https://github.com/microsoft/playwright/issues/9417](url)
Is the `test.info.attachment` solution already implemented?

Thank you in advance.

**Comment by mxschmitt** [comment]
Yes you should be able to do:

```ts
test.info().attachments.push(...)
```

let us know if this works for you!

**Comment by mxschmitt** [comment]
https://github.com/microsoft/playwright/pull/10708 implemented `test.info()` to get the TestInfo object. This gets released in version 1.18 - you can either wait or use our dev release via `npm install @playwright/test@next`.

```ts
test.info().attachments.push(...)
```

**Comment by Mictim** [comment]
@mxschmitt Thank you, it is working for me.
